src/protocol/sort/mod.rs

```diff
@@ -86,3 +86,18 @@ impl AsRef<str> for ComposeStep {
         }
     }
 }
+
+#[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
+pub enum BitPermutationStep {
+    ShareOfOne,
+}
+
+impl Substep for BitPermutationStep {}
+
+impl AsRef<str> for BitPermutationStep {
+    fn as_ref(&self) -> &str {
+        match self {
+            Self::ShareOfOne => "share_of_one",
+        }
+    }
+}
```

Comment:
All of this should be removed. This is not a step that needs to be done. There is already an `r` in place before this protocol is invoked (possibly) and if you generate a new one, it will not validate.